src/coredump/stacktrace.c

```diff
@@ -399,15 +403,124 @@ static int parse_core(int fd, const char *executable, char **ret, JsonVariant **
+                           NULL);
+        if (r < 0) {
+                if (r == -EPROTO) { /* We should have the errno from the child, but don't clobber original error */
+                        int e, k;
+
+                        k = RET_NERRNO(read(error_pipe[0], &e, sizeof(e)));
+                        if (k < 0)
+                                return k;
+                        if (k == sizeof(e))
+                                return e; /* propagate error sent to us from child */
+                        if (k != 0)
+                                return -EIO;
+                }
+
+                return r;
+        }
+        if (r == 0) {
+                _cleanup_(json_variant_unrefp) JsonVariant *package_metadata = NULL;
+                _cleanup_free_ char *buf = NULL;
+
+                (void) setrlimit(RLIMIT_CORE, &RLIMIT_MAKE_CONST(0));
+
+                r = parse_core(fd, executable, ret ? &buf : NULL, ret_package_metadata ? &package_metadata : NULL);
+                if (r < 0)
+                        goto child_fail;
+
+                if (buf) {
+                        _cleanup_fclose_ FILE *out = NULL;
+
+                        out = fdopen(TAKE_FD(return_pipe[1]), "w");
+                        if (!out) {
+                                r = -errno;
+                                goto child_fail;
+                        }
+
+                        fputs(buf, out);
+                        fflush(out);
+                }
+
+                if (package_metadata) {
+                        _cleanup_fclose_ FILE *json_out = NULL;
+
+                        json_out = fdopen(TAKE_FD(json_pipe[1]), "w");
+                        if (!json_out) {
+                                r = -errno;
+                                goto child_fail;
+                        }
+
+                        json_variant_dump(package_metadata, JSON_FORMAT_FLUSH, json_out, NULL);
+                }
+
+                _exit(EXIT_SUCCESS);
+
+        child_fail:
+                (void) write(error_pipe[1], &r, sizeof(r));
+                _exit(EXIT_FAILURE);
+        }
+
+        error_pipe[1] = safe_close(error_pipe[1]);
+        return_pipe[1] = safe_close(return_pipe[1]);
+        json_pipe[1] = safe_close(json_pipe[1]);
+
+        if (ret) {
+                _cleanup_fclose_ FILE *in = NULL;
+
+                in = fdopen(TAKE_FD(return_pipe[0]), "r");
+                if (!in)
+                        return -errno;
+
+                r = read_full_stream(in, ret, NULL);
+                if (r < 0)
+                        return r;
+        }
+
+        if (ret_package_metadata) {
+                _cleanup_fclose_ FILE *json_in = NULL;
+
+                json_in = fdopen(TAKE_FD(json_pipe[0]), "r");
+                if (!json_in)
+                        return -errno;
+
+                r = json_parse_file(json_in, NULL, 0, ret_package_metadata, NULL, NULL);
+                if (r < 0 && r != -EINVAL)
```

Comment:
why ignore EINVAL, and nothing else? deserves comment at least?

also please debug log errors if you eat them up

Reply:
EINVAL means the input file/stream was empty, which simply means we found nothing